pydatastructs/graphs/adjacency_list.py

```diff
@@ -20,6 +20,7 @@ def __new__(cls, *vertices):
         for vertex in vertices:
             obj.__setattr__(vertex.name, vertex)
         obj.vertices = [vertex.name for vertex in vertices]
+        obj.nodes = [vertex for vertex in vertices]
```

Comment:
Not needed.

Reply:
I had to access data stored in the node, considering it as index for that node. The vertex name would be a string which can't be used as index to identify node in transition matrix.
The present code is a rough implementation. Presently I have considered only `adjacencylist` here, but it can be applied to matrix as well with some modifications. I have considered the possibility of weighted edges, where `edge.value` is equivalent to weight. I will add tests and improve the code quality later.

Reply:
Why do you need the data inside the node/vertex?

Reply:
For now I need something as an index to identify the node. I assumed the data can be used to index.

Reply:
The probability of next node in random walk can be decided based on the weight of the edge or degree according to one of the source you shared. Which one should be implemented.